app/uk/gov/hmrc/selfservicetimetopay/controllers/ArrangementController.scala

```diff
@@ -44,103 +44,96 @@ class ArrangementController(ddConnector: DirectDebitConnector,
   val paymentFrequency = "Calendar Monthly"
   val paymentCurrency = "GBP"
 
-  def eligibilityCheck(ttpSubmission: TTPSubmission)(implicit hc: HeaderCarrier): Future[Result] = {
-    eligibilityConnector.checkEligibility(EligibilityRequest(LocalDate.now(), ttpSubmission.taxpayer.get)).flatMap[Result] {
-      case EligibilityStatus(true, _) => changeScheduleDay(ttpSubmission, LocalDate.now.getDayOfMonth)
-      case EligibilityStatus(_, reasons) =>
-        Logger.info(s"Failed eligibility check because: $reasons")
-        Future.successful(Redirect(routes.SelfServiceTimeToPayController.getTtpCallUs()))
-    }
+  private def eligibilityCheck(taxpayer: Taxpayer)(implicit hc: HeaderCarrier) = {
+    eligibilityConnector.checkEligibility(EligibilityRequest(LocalDate.now(), taxpayer))
   }
 
-  def determineMisalignment: Action[AnyContent] = AuthorisedSaUser {
-    implicit authContext => implicit request =>
-      authorizedForSsttp {
+  def determineMisalignment: Action[AnyContent] = AuthorisedSaUser { implicit authContext =>
+    implicit request =>
+      authorizedForSsttp { submission =>
```

Comment:
authorizedForSsttp now provides the TTPSubmission data rather than having to be retrieved via a sessionCache get operation.